WalletWasabi.Fluent.Generators/UIContextGenerator.cs

```diff
@@ -0,0 +1,332 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+
+namespace WalletWasabi.Fluent.Generators;
+
+[Generator]
+public class UiContextGenerator : ISourceGenerator
+{
+	private static string[] Exclusions =
+		new[]
+		{
+			"RoutableViewModel"
+		};
+
+	public void Initialize(GeneratorInitializationContext context)
+	{
+		context.RegisterForSyntaxNotifications(() => new SyntaxReceiver());
+	}
+
+	public void Execute(GeneratorExecutionContext context)
+	{
+		if (context.SyntaxContextReceiver is not SyntaxReceiver receiver)
+		{
+			return;
+		}
+
+		var ctors =
+			ProcessViewModels(context, receiver.ClassDeclarations)
+				.OrderBy(x => x.Identifier.ValueText)
+				.ToList();
```

Comment:
Not an important thing, but `ToArray` in that case won't allocate a wrapper like `List<T>`. LINQ received at some point an ability to determine size of linked iterators, so if one can tell its size, it happens.

For example, `Select` doesn't change size of the original source, but maps elements. Then an operator like `OrderBy` or `ToArray` can ask for a size hint to allocate an array without any overhead.